oap-logstream/src/main/java/oap/logstream/disk/DiskLoggerBackend.java

```diff
@@ -132,19 +153,26 @@ public void log( ProtocolVersion protocolVersion, String hostName, String filePr
 
         Metrics.counter( "logstream_logging_disk_counter", List.of( Tag.of( "from", hostName ) ) ).increment();
         Metrics.summary( "logstream_logging_disk_buffers", List.of( Tag.of( "from", hostName ) ) ).record( length );
-        var writer = writers.get( new LogId( filePreffix, logType, hostName, shard, properties, headers, types ) );
-        log.trace( "logging {} bytes to {}", length, writer );
-        try {
-            writer.write( protocolVersion, buffer, offset, length, this.listeners::fireError );
-        } catch( Exception e ) {
-            var headersWithTypes = new ArrayList<String>();
-            for( int i = 0; i < headers.length; i++ ) {
-                headersWithTypes.add( headers[i] + " [" + Lists.map( List.of( ArrayUtils.toObject( types[i] ) ), oap.template.Types::valueOf ) + "]" );
-            }
+        List<AbstractWriter<? extends Closeable>> writerList = writers.get( new LogId( filePreffix, logType, hostName, shard, properties, headers, types ) );
```

Comment:
`writerList` for 2 formats, but `LogId` identifies only one `filePreffix` and `logType`.
Should it be changed? Possibly 2 `LogId` should be transferred

Reply:
LogId#logType is not a log file format, it is a data format (bids, impressions)